docs/user_guide/backend_configuration.rst

```diff
@@ -0,0 +1,286 @@
+Backend Configuration
+=====================
+
+NeuroConv offers highly convenient control over the type of file backend and the way its datasets are configured.
+
+Find out more about possible file types in the `main NWB documentation <https://nwb-overview.readthedocs.io/en/latest/faq_details/why_hdf5.html#why-use-hdf5-as-the-primary-backend-for-nwb>`_.
+
+Find out more about chunking and compression in the `advanced NWB tutorials for dataset I/O settings <https://pynwb.readthedocs.io/en/stable/tutorials/advanced_io/h5dataio.html#sphx-glr-tutorials-advanced-io-h5dataio-py>`_.
+
+Find out more about memory buffering of large source files in the `advanced NWB tutorials for iterative data write <https://pynwb.readthedocs.io/en/stable/tutorials/advanced_io/plot_iterative_write.html#sphx-glr-tutorials-advanced-io-plot-iterative-write-py>`_.
+
+
+
+Default configuration
+---------------------
+
+To retrieve a default configuration for an in-memory ``pynwb.NWBFile`` object, use the ``get_default_backend_configuration`` function...
+
+.. code-block:: python
+
+    from datetime import datetime
+    from uuid import uuid4
+
+    from dateutil import tz
+    from neuroconv.tools.nwb_helpers import get_default_backend_configuration
+    from pynwb import NWBFile, TimeSeries
+
+    session_start_time = datetime(2020, 1, 1, 12, 30, 0, tzinfo=tz.gettz("US/Pacific"))
+    nwbfile = NWBFile(
+        identifier=str(uuid.uuid4()),
+        session_start_time=session_start_time,
+        session_description="A session of my experiment.",
+    )
+
+    time_series = TimeSeries(
+        name="MyTimeSeries",
+        description="A time series from my experiment.",
+        unit="cm/s",
+        data=[1., 2., 3.],
+        timestamps=[0.0, 0.2, 0.4],
+    )
+    nwbfile.add_acquisition(time_series)
+
+    default_backend_configuration = get_default_backend_configuration(
+        nwbfile=nwbfile, backend="hdf5"
+    )
+
+From which a printout of the contents looks like...
+
+.. code-block:: python
+
+    print(default_backend_configuration)
+
+.. code-block:: bash
+
+    Configurable datasets identified using the hdf5 backend
+    -------------------------------------------------------
+
+    acquisition/MyTimeSeries/data
+    -----------------------------
+      dtype : float64
+      full shape of source array : (3,)
+      full size of source array : 0.00 GB
```

Comment:
I'm not crazy about printing sizes like this. I'll propose a small PR to print size with the appropriate units.

Reply:
A planned eventual follow-up is the auto-scaling into 'reasonable' units - feel free to take that if you want (I know I've seen you do that kind of logarithm based stuff before)

Reply:
But yes, it's independent of this PR which is just to get basic docs up

Reply:
Also planning a follow-up with HTML repr for analogous usage in notebooks, was going to let @h-mayorquin handle that due to their growing familiarity with the approach